gindent: Fix type incompatibility introduced with gperf 3.1

The patch is based on the idea proposed by gnw3. I tried this on a clean
build with macports 2.4.1 on El Capitan 10.11.6
The problem was introduced with gperf 3.1 resulting in a build
failure for gindent.
see: https://trac.macports.org/ticket/54466
Index: src/lexi.c
===============================================================================
--- src/lexi.c.orig	2017-08-27 18:33:16.000000000 +0200
+++ src/lexi.c	2017-08-27 18:35:22.000000000 +0200
@@ -201,7 +201,7 @@
 #ifdef __GNUC__
 __inline
 #endif
-templ_ty *is_reserved (const char *str, unsigned int len);
+templ_ty *is_reserved (const char *str, size_t len);
 
 #include "gperf.c"
 
@@ -215,7 +215,7 @@
 #ifdef __GNUC__
 __inline
 #endif
-templ_ty *is_reserved_cc (register const char *str, register unsigned int len);
+templ_ty *is_reserved_cc (register const char *str, register size_t len);
 
 #include "gperf-cc.c"
 
